src/System.Private.ServiceModel/tools/scripts/WcfTestServerSetup.cmd

```diff
@@ -0,0 +1,191 @@
+REM call :Run icacls %CurrentRepo%\src\System.Private.ServiceModel\tools\PRService /grant:r "IIS APPPOOL\%PRServiceName%":RX /T /Q
+REM if ERRORLEVEL 1 goto :Failure
+
+:SkipPRService
+
+:: Grant master PR service's app pool access to current repo if master PRService exists
+if /I '%MasterRepo%'=='%CurrentRepo%' goto :SkipGrantAccess
+if NOT EXIST %MasterRepo% goto :SkipGrantAccess
+%appcmd% list apppool %PRServiceMaster% >nul
+if ERRORLEVEL 1 goto :SkipGrantAccess
+echo Grant master PRServer app pool: %PRServiceMaster% "Modify" access to %CurrentRepo% and its sudirectories
+call :Run icacls %CurrentRepo% /grant:r "IIS APPPOOL\%PRServiceMaster%":M /T /Q
+if ERRORLEVEL 1 goto :Failure
+
+:SkipGrantAccess
+
+:: Install certificates if %WcfTestDir% does not exist; otherwise, assume certs are installed already
+if EXIST %WcfTestDir% goto :SkipCertInstall
+
+:: Make sure there is an HTTPS binding in IIS site "Default Web Site"
+%appcmd% list site "Default Web Site" /text:* | findstr /i bindings | findstr /i https >nul
+if ERRORLEVEL 1 (
+    echo Add an HTTPS binding to "Default Web Site"
+    call :Run %appcmd% set site "Default Web Site" /+bindings.[protocol='https',bindingInformation='*:443:']
+    if ERRORLEVEL 1 goto :Failure
+)
+
+:: Use current repo for cert gen if master repo does not exist
+if NOT EXIST %MasterRepo% (
+    set MasterRepo=%CurrentRepo%
+    set WcfServiceMaster=%WcfServiceName%
+)
+
+echo Build CertificateGenerator tool
+call :Run %MasterRepo%\src\System.Private.ServiceModel\tools\scripts\BuildCertUtil.cmd
+if ERRORLEVEL 1 goto :Failure
+
+echo Run CertificateGenerator tool. This will take a little while...
+md %WcfTestDir%
+set certGen=%MasterRepo%\bin\Wcf\tools\CertificateGenerator\CertificateGenerator.exe
+echo ^<?xml version="1.0" encoding="utf-8"?^>^<configuration^>^<appSettings^>^<add key="testserverbase" value="%WcfServiceMaster%"/^>^<add key="CertExpirationInDay" value="%CertExpirationInDays%"/^>^<add key="CrlFileLocation" value="%WcfTestDir%\test.crl"/^>^</appSettings^>^<startup^>^<supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5"/^>^</startup^>^</configuration^>>%certGen%.config
+call :Run %certGen%
+if ERRORLEVEL 1 goto :Failure
+
+echo Configue SSL certificate ports
+call :Run powershell -NoProfile -ExecutionPolicy unrestricted %MasterRepo%\src\System.Private.ServiceModel\tools\scripts\ConfigHttpsPort.ps1
+if ERRORLEVEL 1 goto :Failure
+
+:: TODO: Grant all existing app pools named WCFService# 'Read' access to %WcfTestDir%. This is not needed for now
+
+:SkipCertInstall
+
+:: Grant app pool %WcfServiceName% "Read" access to %WcfTestDir% and its subdirectories
+echo Grant app pool %WcfServiceName% "Read" access to %WcfTestDir% and its subdirectories
+call :Run icacls %WcfTestDir% /grant:r "IIS APPPOOL\WcfService%Id%":R /T /Q
+if ERRORLEVEL 1 goto :Failure
+
+:: Clean up log file if everything worked perfectly
```

Comment:
To be clear, I don't support logging by this script at all. I just use a file to temporarily save output of each command currently running and show it when it's a failure. Even if I don't delete it, you only see the output from last command.

Supporting logging involves fairly amount of work. I don't have plan to do it and I don't think we should spend time to do it.